File: checkout-ruby/lib/promotional_rule.rb
class PromotionalRule

  def get_discount_entity_type
    raise "Implementation necessary"
  end

  def get_rule_discount_entity
    raise "Implementation necessary"
  end

  def is_eligible_for_discount?(entity)
    entity >= get_rule_discount_entity
  end

  attr_accessor :value_type

end

File: checkout-java/src/main/java/com/vinterior/checkout/models/PromotionalRule.java
package com.vinterior.checkout.models;

import com.vinterior.checkout.RuleDiscountValue;
import com.vinterior.checkout.enums.RuleDiscountEntityType;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public abstract class PromotionalRule {

    abstract RuleDiscountEntityType getDiscountEntityType();

    protected RuleDiscountValue valueType;

    abstract double getRuleDiscountEntity();

    public boolean isEligibleForDiscount(double entity) {
        return entity >= getRuleDiscountEntity();
    }

}
